### PR TITLE
Decouple gossip

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

N/A

## Proposed Changes

Start decoupling gossip logic according to https://github.com/ethereum/consensus-specs/pull/3244

### CODE REVIEW

File: beacon_node/http_api/src/publish_blocks.rs [hunk 1]
```diff
@@ -32,51 +31,33 @@ pub async fn publish_block<T: BeaconChainTypes>(
...
-                    "no blob cached for block"
-                )));
-            }
-        } else {
-            crate::publish_pubsub_message(network_tx, PubsubMessage::BeaconBlock(block.clone()))?;
-            block.into()
-        };
+    crate::publish_pubsub_message(network_tx, PubsubMessage::BeaconBlock(block.clone()))?;
+
+    /* TODO: publish all blob sidecars associated with this block */
```
Comment (jimmygchen) [inline beacon_node/http_api/src/publish_blocks.rs:36]: there's an open discussion whether to do blob publishing separately
https://github.com/ethereum/beacon-APIs/issues/300
PR: https://github.com/ethereum/beacon-APIs/pull/302

**Comment by realbigsean** [comment]
@pawanjay176  would it be helpful if we rebase this on the main blob branch and merge it?

**Comment by pawanjay176** [comment]
I think this will overlap with some of the caching stuff we are doing. But I can cherry pick the gossip validation changes in this PR and we can continue with the caching stuff in another PR.

**Comment by pawanjay176** [comment]
Relevant work extracted into #4092

**Pull request closed without merging**